src/main/java/bio/terra/service/snapshot/SnapshotDao.java

```diff
@@ -304,33 +304,108 @@ public Snapshot retrieveSnapshot(UUID snapshotId, boolean onlyRetrieveAvailable)
   public SnapshotProject retrieveSnapshotProject(UUID snapshotId, boolean onlyRetrieveAvailable) {
     logger.debug("retrieve snapshot id: " + snapshotId);
     String sql =
-        "SELECT snapshot.id, name, snapshot.profile_id, google_project_id, "
-            +
-            // Select the source dataset project information
-            "(SELECT jsonb_agg(ds)\n"
-            + "  FROM (SELECT d.id, d.name, p.profile_id as \"profileId\", p.google_project_id as \"dataProject\"\n"
-            + "        FROM snapshot_source ss, dataset d, project_resource p\n"
-            + "        WHERE ss.dataset_id = d.id"
-            + "        AND d.project_resource_id = p.id"
-            + "        AND snapshot.id = ss.snapshot_id) ds)"
-            + "  AS dataset_sources, "
-            // Detect the cloud provider of the snapshot
+        "SELECT snapshot.id, name, snapshot.profile_id, "
```

Comment:
one of the main reasons to roll this onto a select statement in the first place was to reduce the amount of calls DRS lookups need to make to the db.  You could actually leave this by just doing a left join to the snapshot project table
```suggestion
            "(SELECT jsonb_agg(ds)\n"
            + "  FROM (SELECT d.id, d.name, p.profile_id as \"profileId\", p.google_project_id as \"dataProject\"\n"
            + "        FROM snapshot_source ss JOIN dataset d ON  ss.dataset_id = d.id"
            + "           LEFT JOIN project_resource p ON d.project_resource_id = p.id"
            + "        WHERE snapshot.id = ss.snapshot_id) ds)"
            + "  AS dataset_sources, "
            + "FROM snapshot "
            + "LEFT JOIN project_resource ON snapshot.project_resource_id = project_resource.id "
            + "WHERE snapshot.id = :id";
```

Reply:
That makes sense! That's so much nicer!!!

Reply:
We still want to detect the cloud provider right?